github/GithubIntegration.py

```diff
@@ -263,7 +267,18 @@ def get_access_token(
         if not isinstance(permissions, dict):
             raise GithubException(status=400, data={"message": "Invalid permissions"}, headers=None)
 
-        body = {"permissions": permissions}
+        if not isinstance(repositories, (list, type(None))):
+            raise GithubException(status=400, data={"message": "Invalid repositories"}, headers=None)
+
+        if not isinstance(repository_ids, (list, type(None))):
+            raise GithubException(status=400, data={"message": "Invalid repository_ids"}, headers=None)
+
+        body: dict[str, Any] = {"permissions": permissions}
+        if repositories is not None:
+            body["repositories"] = repositories
+        if repository_ids is not None:
+            body["repository_ids"] = repository_ids
```

Comment:
Here we would have to populate the two fields from the single `repositories` list based on element types using list comprehension. I presume empty lists are allowed in the request (please check with docu and test by calling Github API).